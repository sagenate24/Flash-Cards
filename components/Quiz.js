import React, { Component } from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  TouchableWithoutFeedback,
  Platform,
} from 'react-native';
import { connect } from 'react-redux';
import { white, red, lightBlue } from '../utils/colors';

// import IOSCard from './IOSCard';
// import AndriodCard from './AndriodCard';
import Card from './Card';
// import FlipTest from './FlipTest';

class Quiz extends Component {
  state = {
    questionIndex: 0,
    correctAnswers: 0,
  };

  correctAnswer = () => {
    const { currentDeck } = this.props;
    const { questions } = currentDeck;
    const { questionIndex, correctAnswers } = this.state;

    if (questionIndex + 1 === questions.length) {
      this.goToResults('correct');
    } else {
      this.setState(() => ({
        correctAnswers: correctAnswers + 1,
        questionIndex: questionIndex + 1,
      }));
    }
  };

  wrongAnswer = () => {
    const { currentDeck } = this.props;
    const { questionIndex } = this.state;
    const { questions } = currentDeck;

    if (questionIndex + 1 === questions.length) {
      this.goToResults('incorrect');
    } else {
      this.setState(() => ({ questionIndex: questionIndex + 1 }));
    }
  };

  goToResults = (lastAnswer) => {
    const { correctAnswers } = this.state;
    const { navigation } = this.props;

    navigation.navigate(
      'Results',
      {
        correctAnswers: lastAnswer === 'correct' ? correctAnswers + 1 : correctAnswers,
        currentDeck: this.props.currentDeck,
      },
    );

    this.setState(() => ({
      questionIndex: 0,
      correctAnswers: 0,
    }));
  };

  render() {
    const { currentDeck } = this.props;
    const { questions } = currentDeck;
    const { questionIndex } = this.state;
    const questionsRemaining = questions.length - questionIndex;

    return (
      <View style={styles.container}>
        <View style={{ flex: 4 }}>
          <Card question={questions[questionIndex].question} answer={questions[questionIndex].answer} questionsRemaining={questionsRemaining} questionIndex={questionIndex} />
        </View>
        <View style={[styles.bottomContent, { flex: 1 }]}>
          <TouchableOpacity
            style={styles.correctBtn}
            onPress={this.correctAnswer}
          >
            <Text style={styles.btnText}>Correct</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.incorrectBtn}
            onPress={this.wrongAnswer}
          >
            <Text style={styles.btnText}>Incorrect</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  bottomContent: {
    marginTop: 20,
    marginLeft: 20,
    marginRight: 20,
  },
  correctBtn: {
    backgroundColor: lightBlue,
    padding: 10,
    borderRadius: 2,
  },
  incorrectBtn: {
    backgroundColor: red,
    padding: 10,
    marginTop: 20,
    borderRadius: 2,
  },
  btnText: {
    fontSize: 22,
    fontWeight: 'bold',
    color: white,
    textAlign: 'center',
  },
});

function mapStateToProps(state, { navigation }) {
  const { currentDeck } = navigation.state.params;

  return {
    currentDeck,
  };
}

export default connect(mapStateToProps)(Quiz);
