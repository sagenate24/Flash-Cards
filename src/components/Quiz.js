import React, { Component } from 'react';
import { View, Platform } from 'react-native';
import { connect } from 'react-redux';
import { red, queenBlue, white } from '../utils/colors';

import Card from './Card';
import NASBtn from './NASBtn';

class Quiz extends Component {
  state = {
    questionIndex: 0,
    correctAnswers: 0,
  };

  correctAnswer = () => {
    const { deck } = this.props;
    const { questions } = deck;
    const { questionIndex, correctAnswers } = this.state;

    if (questionIndex + 1 === questions.length) {
      this.goToResults('correct');
    } else {
      this.setState(() => ({
        correctAnswers: correctAnswers + 1,
        questionIndex: questionIndex + 1,
      }));
    }
  };

  incorrectAnswer = () => {
    const { deck } = this.props;
    const { questions } = deck;
    const { questionIndex } = this.state;

    if (questionIndex + 1 === questions.length) {
      this.goToResults('incorrect');
    } else {
      this.setState(() => ({ questionIndex: questionIndex + 1 }));
    }
  };

  goToResults = (lastAnswer) => {
    const { correctAnswers } = this.state;
    const { navigation, deck } = this.props;

    navigation.navigate(
      'Results',
      {
        correctAnswers: lastAnswer === 'correct' ? correctAnswers + 1 : correctAnswers,
        currentDeck: deck,
      },
    );

    this.setState(() => ({
      questionIndex: 0,
      correctAnswers: 0,
    }));
  };

  render() {
    const { deck } = this.props;
    const { questions } = deck;
    const { questionIndex } = this.state;
    const questionsRemaining = questions.length - questionIndex;

    return (
      <View style={{ flex: 1, padding: 20 }}>
        <View style={Platform.OS === 'ios' ? { flex: 4.2 } : { flex: 3.2 }}>
          <Card
            questionsRemaining={questionsRemaining}
            answer={questions[questionIndex].answer}
            question={questions[questionIndex].question}
          />
        </View>
        <View style={{ flex: 1 }}>
          <NASBtn
            onPress={this.correctAnswer}
            textStyle={{ color: queenBlue }}
            tintColor={{ borderColor: queenBlue, borderWidth: 3, backgroundColor: white }}
          >
            Correct
          </NASBtn>
          <NASBtn
            onPress={this.incorrectAnswer}
            textStyle={{ color: red }}
            tintColor={{ borderColor: red, borderWidth: 3, backgroundColor: white }}
          >
            Incorrect
          </NASBtn>
        </View>
      </View>
    );
  }
}

function mapStateToProps(state, { navigation }) {
  const { currentDeck } = navigation.state.params;

  return {
    deck: currentDeck,
  };
}

export default connect(mapStateToProps)(Quiz);
