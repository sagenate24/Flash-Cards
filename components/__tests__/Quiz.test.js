import 'react-native';
import React from 'react';
import Quiz from '../Quiz';
import renderer from 'react-test-renderer';

const mockCurrentDeck = {
    title: 'React',
    timeStamp: 1534284894237,
    recentScore: 35,
    questions: [
      {
        question: 'What is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is React?',
        answer: 'A library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfaces',
      },
      {
        question: 'Where do you make Ajax requests in React?',
        answer: 'The componentDidMount lifecycle event',
      },
    ],
}

test('Quiz snapShot', () => {
  const snap = renderer.create(
    <Quiz.WrappedComponent currentDeck={mockCurrentDeck} />
  ).toJSON();

  expect(snap).toMatchSnapshot();
});