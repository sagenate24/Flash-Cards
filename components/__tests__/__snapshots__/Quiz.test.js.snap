// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz snapShot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 4,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={undefined}
      testID={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backfaceVisibility": "hidden",
            "backgroundColor": "#fff",
            "height": "100%",
            "opacity": 90,
            "padding": 20,
            "shadowColor": "rgba(0,0,0,0.24)",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "transform": Array [
              Object {
                "rotateY": "0deg",
              },
            ],
          }
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "justifyContent": "flex-start",
                "transform": Array [
                  Object {
                    "scale": 0.1,
                  },
                ],
              }
            }
          >
            2
             Left
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "opacity": 0.6,
            }
          }
        >
          Question
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "90%",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#404040",
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            What is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is React?
          </Text>
        </View>
      </View>
      <View
        collapsable={undefined}
        style={
          Object {
            "backfaceVisibility": "hidden",
            "backgroundColor": "#fff",
            "height": "100%",
            "left": 0,
            "opacity": -89,
            "padding": 20,
            "position": "absolute",
            "shadowColor": "rgba(0,0,0,0.24)",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "top": 0,
            "transform": Array [
              Object {
                "rotateY": "180deg",
              },
            ],
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "justifyContent": "flex-start",
                "transform": Array [
                  Object {
                    "scale": 0.1,
                  },
                ],
              }
            }
          >
            2
             Left
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "opacity": 0.6,
            }
          }
        >
          Answer
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "90%",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#404040",
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            A library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfaces
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "padding": 20,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#a8dadc",
          "borderRadius": 2,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Correct
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e63946",
          "borderRadius": 2,
          "marginTop": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Incorrect
      </Text>
    </View>
  </View>
</View>
`;
